tests/test_profile.py

```diff
@@ -0,0 +1,170 @@
+import pytest
+
+import json
+import os
+import shutil
+
+from great_expectations.profile.base import DataSetProfiler
+from great_expectations.profile.basic_dataset_profiler import BasicDatasetProfiler
+from great_expectations.profile.columns_exist import ColumnsExistProfiler
+from great_expectations.dataset.pandas_dataset import PandasDataset
+from great_expectations.data_context import DataContext
+from great_expectations.data_context.util import safe_mmkdir
+
+# Tests to write:
+# test_cli_method_works  -> test_cli
+# test context-based profile methods
+# test class-based profile methods
+
+
+def test_DataSetProfiler_methods():
+    toy_dataset = PandasDataset({"x": [1, 2, 3]})
+
+    assert DataSetProfiler.validate_dataset(1) == False
+    assert DataSetProfiler.validate_dataset(toy_dataset)
+
+    with pytest.raises(NotImplementedError) as e_info:
+        DataSetProfiler.profile(toy_dataset)
+
+
+def test_ColumnsExistProfiler():
+    toy_dataset = PandasDataset({"x": [1, 2, 3]})
+    expectations_config, evr_config = ColumnsExistProfiler.profile(toy_dataset)
+
+    print(json.dumps(expectations_config, indent=2))
+
+    # assert expectations_config == {
+    #     "data_asset_name": None,
+    #     "data_asset_type": "Dataset",
+    #     "meta": {
+    #         "great_expectations.__version__": "0.7.0-beta",
+    #         "ColumnsExistProfiler": {
+    #             "created_by": "BasicDatasetProfiler",
+    #             "created_at": 0,
+    #         },
+    #     },
+    #     "expectations": [
+    #         {
+    #             "expectation_type": "expect_column_to_exist",
+    #             "kwargs": {
+    #                 "column": "x"
+    #             }
+    #         }
+    #     ]
+    # }
+
+
+def test_BasicDatasetProfiler():
+    toy_dataset = PandasDataset({"x": [1, 2, 3]})
+    assert len(toy_dataset.get_expectations(
+        suppress_warnings=True)["expectations"]) == 0
+
+    expectations_config, evr_config = BasicDatasetProfiler.profile(toy_dataset)
+
+    # print(json.dumps(expectations_config, indent=2))
+
+    assert len(toy_dataset.get_expectations(
+        suppress_warnings=True)["expectations"]) > 0
+
+    # We should add an additional test that instantiates the batch via context, so the data_asset_name will be populated.
+    assert expectations_config["data_asset_name"] == None
+    assert "BasicDatasetProfiler" in expectations_config["meta"]
+    # We should add an additional test that instantiates the batch via context, so that batch_kwargs will be populated.
+    assert set(expectations_config["meta"]["BasicDatasetProfiler"].keys()) == {
+        "created_by", "created_at"
+    }
+    for exp in expectations_config["expectations"]:
+        assert "BasicDatasetProfiler" in exp["meta"]
+        assert exp["meta"]["BasicDatasetProfiler"] == {
+            "confidence": "very low"
+        }
+
+    # Example:
```

Comment:
Would be nice to have a test of the profiler's behavior: i.e. does it find what it's supposed to. Run it on Titanic.csv and check for consistency?

Reply:
Yes, I just don't want to lock that in until the profiler is in a stable place.

Reply:
kk